### PR TITLE
Make it possible to set bot_prompt query param for line

### PR DESCRIPTION
**Pull request by k-yomo**
To display the option to add your LINE Official Account as a friend, we need to pass `bot_prompt` query parameter for the auth url.
This PR adds `SetBotPrompt` function to set the query param.

details for the `bot_promt` is written in the document below.
https://developers.line.biz/en/docs/line-login/link-a-bot/#redirect-users

bot-prompt 1dd4e585

### CODE REVIEW

**Review by techknowlogick — APPROVED** [review]

**Review by bentranter — APPROVED** [review]
LGTM! (pending CI)

File: providers/line/line_test.go
Comment (bentranter) [inline providers/line/line_test.go:60]: Looks like this bit of the test can be flaky depending on the order of the query params,

```
 --- FAIL: Test_SetBotPrompt (0.00s)
    line_test.go:60: 
        	Error Trace:	line_test.go:60
        	Error:      	"https://access.line.me/oauth2/v2.1/authorize?bot_prompt=normal&client_id=&redirect_uri=%2Ffoo&response_type=code&state=test_state" does not contain "&bot_prompt=normal"
        	Test:       	Test_SetBotPrompt
```

for example, it may create the URL param prefixed with `?` or `&`. It might be better not to check that leading character, and just do,

```go
a.Contains(s.AuthURL, "bot_prompt=normal")
```

**Review by bentranter — CHANGES_REQUESTED** [review]
One comment about the tests, after that it should be good to merge. Thanks for the contribution @k-yomo!

File: providers/line/line_test.go
Comment (k-yomo) [reply providers/line/line_test.go]: Thanks for the good catch! fixed to do `a.Contains(s.AuthURL, "bot_prompt=normal")`!

**Review by bentranter — APPROVED** [review]
LGTM, thanks @k-yomo!

**Pull request merged**
